rag/advanced_rag/harness/tools/navigation.py

```
@@ -360,94 +330,203 @@ async def mindmap_navigate(tools, topic: str, keywords: str = "", doc_scope: lis
+            continue
+        seen_nodes.add(node_key)
         try:
-            hits = await search_dataset_nav(
-                kb.tenant_id,
-                kb.id,
-                query,
-                embd_mdl=getattr(tools, "embed_mdl", None),
-                top_k=_NAV_MAX_HITS_PER_KB,
-            )
+            ok, data = await dataset_api_service.list_nav_children(kb.id, kb.tenant_id, name, page=1, page_size=_NAV_CHILDREN_PAGE_SIZE)
         except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)